src/gurt/dlog.c

```diff
@@ -416,9 +417,20 @@ void d_vlog(int flags, const char *fmt, va_list ap)
 			 tm->tm_hour, tm->tm_min, tm->tm_sec,
 			 (long int) tv.tv_usec / 10000, mst.uts.nodename);
 
-	if (mst.oflags & DLOG_FLV_TAG)
-		hlen += snprintf(b + hlen, sizeof(b) - hlen,
-				 "%s ", d_log_xst.tag);
+	if (mst.oflags & DLOG_FLV_TAG) {
+		if (mst.oflags & DLOG_FLV_LOGPID) {
+			static __thread pid_t tid = -1;
+
+			if (tid == -1)
```

Comment:
hmm, why check -1 after assignment?

Reply:
to only do gettid() syscall once per pthread